README.md

```diff
@@ -1,19 +1,153 @@
-![CI](https://github.com/Cardinal-Cryptography/rush/workflows/CI/badge.svg)
+[![Crate][crate-image]][crate-link]
+[![Docs][docs-image]][docs-link]
+[![Build Status][build-image]][build-link]
+[![Apache 2.0 Licensed][license-image]][license-link]
+![Rust Stable][rustc-image]
 
-# rush
+### Aleph Consensus
 
-rust implementation of Aleph protocol
+![aleph logo](logo.jpg "Aleph logo")
 
-# examples
+### Overview
 
-There is a dummy implementation of an honest committee member that is not
+Aleph is an asynchronous and Byzantine fault tolerant consensus protocol aimed
+at ordering arbitrary messages (transactions). It has been designed to operate
+continuously under conditions where there is no bound on message-delivery delay
+and under the assumption that there is a significant probability of malicious
+behavior, making it an excellent fit for blockchain-related applications.
+For more information, check [the paper][paper-link]
+
+This repository contains a Rust implementation of Aleph that offers a convenient
+API enabling seamless application to various problems. The prime application of
+the repository is the consensus engine (sometimes called the "finality gadget")
+of the [Aleph Zero blockchain][aleph-node-link].
+
+### Detailed documentation
+
+If the crate's [documentation][docs-link] seems to be not comprehensive enough,
+please refer to the [detailed version][reference-link].
+
+### Implementation status
+
+- current version is asynchronous, so it's driven by consensus events as opposed
+  to some clock ticks
+- while being asynchronous, the performance is still optimal in partially
+  synchronous environment
+- guaranteed safety in partially synchronous environment
```

Comment:
- guaranteed safety even in asynchronous environment

Reply:
Changed.